### PR TITLE
Add remote-protocol at TcpClients

### LINKED ISSUE #344: Additional protocol attribute at TcpClient
An additional protocol attribute shall be added to the TcpClient for indicating, whether http or https has to be used for consuming some operation.

**Required Changes:**
OAS
- [x] attribute and enumeration at 'tcp client' in control-construct:
- [x] attribute and enumeration at 'tcp client' in common-components
- [x] attribute and enumeration at /core-model-1-4:control-construct/logical-termination-point={uuid}/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol:
- [x] additional field in service body + examples:
  - [x] /v1/bequeath-your-data-and-die:
  - [x] PromptForBequeathingDataCausesRequestForBroadcastingInfoAboutServerReplacement
  - [x] /v1/register-yourself:
  - [x] PromptForRegisteringCausesRegistrationRequest
  - [x] /v1/embed-yourself:
  - [x] PromptForEmbeddingCausesRequestForBequeathingData
  - [x] /v1/redirect-service-request-information
  - [x] /v1/redirect-oam-request-information:
  - [x] /v1/inquire-oam-request-approvals:
  - [x] /v1/update-client:
  - [x] /v1/redirect-topology-change-information:

CONFIGfile
- [x] all tcp-clients

ClassDiagram
- [x] Update TcpClient in ClassDiagram

### LINKED ISSUE #355: Replacing 'https' and enclosing URL in callbacks with quotes
As per https://github.com/openBackhaul/ApplicationPattern/issues/331 and  https://github.com/openBackhaul/ApplicationPattern/issues/344, an additional attribute called "_local-protocol_" and "_remote-protocol_ are added at TcpServer and TcpClient instances respectively for dynamically choosing http or https while consuming operations.

Required changes:

OAS -  Callback URL:
- [x] the protocol "https" in URL of callbacks to be replaced by actual reference to protocol 
- [x] enclosing the URL at callbacks with quotes for avoiding syntax error

### LINKED ISSUE #356: Moving parameter section in callback to post section
Problem statement:

Currently, "parameter" section inside callback definition is placed above "post" section. While generating stub in swagger,  we could see an improper structure generated (we have both post and parameters having URL references and actual parameters section is empty)

Solution:
Moving "parameter" section in callback definition under "post" section would generate swagger UI as expected

### PR DESCRIPTION
**Pull request by openBackhaul**
Find list of changes in issue #344

Fixes #344
Fixes #355
Fixes #356

### CODE REVIEW

**Review by PrathibaJee — COMMENTED** [review]
.

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
> Couldnt able to comment in the specific lines , thats why providing it as a common comment.

- **ApplicationPattern.yaml :**
      Overall , the  file looks good. All the services got complemented with an attribute for protocol with proper reference to the tcp-server/client protocol, description and examples.
      _**Comment :**_ 
      for all the callbacks , currently , the protocols are starting with https://
      https://[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp.......
      Is this is fine or this needs to be a reference to the remote-protocol? Kindly let me know.

- **ApplicationPattern+data.json**
      _**Comment :**_ 
      In this file , in all the action-profile-configuration , require to change the protocol-reference from
      /tcp-server-interface-configuration/protocol
      to
      /tcp-server-interface-configuration/local-protocol

- **ApplicationPattern+profileInstances.yaml :**
      _**Comment :**_ 
      In this file , for the protocol-reference attribute ofthe action profiles *-action-p-000, *-action-p-001, *-action-p-002 require to modify the protocol-reference to point to the "local-protocol"

- **ApplicationPattern+services.yaml**
      _**Comment :**_ 
      In this file , whether we have to include the protocol attribute for the tcp-clients too ? Kindly let me know. Thanks.

File: ApplicationPattern.yaml [hunk 1]
```diff
@@ -7984,17 +8107,17 @@ paths:
...
-                - tcp-server-interface-1-0:protocol
+                - tcp-server-interface-1-0:local-protocol
               properties:
-                tcp-server-interface-1-0:protocol:
+                tcp-server-interface-1-0:local-protocol:
                   type: string
                   enum:
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_HTTP'
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_HTTPS'
                     - 'tcp-server-interface-1-0:PROTOCOL_TYPE_NOT_YET_DEFINED'
```
[the remark this answers is not in the document]
Comment (openBackhaul) [reply ApplicationPattern.yaml:8117]: Same reason as in 5G-xhaul. If device does not support HTTP or HTTPS the management interface has to represent the default value. Just ignore it.

**Comment by openBackhaul** [comment]
In OAS at callbacks:
Replaced _https_ by _[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol]_

In CONFIGfile:
Changed from _/tcp-server-interface-configuration/protocol_ to _/tcp-server-interface-configuration/local-protocol_

In ProfileInstanceList:
Changed from _/tcp-server-interface-configuration/protocol_ to _/tcp-server-interface-configuration/local-protocol_

In ServiceList:
_protocol_ attribute added to TcpClients

File: ApplicationPattern.yaml [hunk 2]
```diff
@@ -600,6 +636,7 @@ paths:
                         - application-release-number
                         - embedding-operation
                         - client-update-operation
+                        - application-protocol
```
Comment (IswaryaaS) [inline ApplicationPattern.yaml:639]: line 620 : callback uri definition for PromptForRegisteringCausesRegistrationRequest has protocol https which needs to be updated to reference registry-office-protocol 
{$request.body#registry-office-protocol}://...

Comment (IswaryaaS) [inline ApplicationPattern.yaml:639]: line 620 : callback uri definition for PromptForRegisteringCausesRegistrationRequest has protocol https which needs to be updated to reference registry-office-protocol 
{$request.body#registry-office-protocol}://...

File: ApplicationPattern.yaml [hunk 3]
```diff
@@ -1067,6 +1146,7 @@ paths:
                 service-log-application: 'ExecutionAndTraceLog'
                 service-log-application-release-number: '2.0.0'
                 service-log-operation: '/v1/record-service-request'
+                service-log-protocol: 'HTTP'
```
Comment (IswaryaaS) [inline ApplicationPattern.yaml:1149]: line 1201: callback uri definition for ServiceRequestCausesLoggingRequest has protocol https which needs to be updated to reference service-log-protocol 
{$request.body#service-log-protocol}://...

File: ApplicationPattern.yaml [hunk 4]
```diff
@@ -1308,6 +1397,7 @@ paths:
                 oam-log-application: 'OamLog'
                 oam-log-application-release-number: '2.0.0'
                 oam-log-operation: '/v1/record-oam-request'
+                oam-log-protocol: 'HTTP'
```
Comment (IswaryaaS) [inline ApplicationPattern.yaml:1400]: line 1452: callback uri definition for OamRequestCausesLoggingRequest has protocol https which needs to be updated to reference oam-log-protocol {$request.body#oam-log-protocol}://...

File: ApplicationPattern.yaml [hunk 5]
```diff
@@ -1632,6 +1731,7 @@ paths:
                 oam-approval-application: 'AdministratorAdministration'
                 oam-approval-application-release-number: '2.0.0'
                 oam-approval-operation: '/v1/approve-oam-request'
+                oam-approval-protocol: 'HTTP'
```
Comment (IswaryaaS) [inline ApplicationPattern.yaml:1734]: line 1786: callback uri definition for OamRequestCausesInquiryForAuthentication has protocol https which needs to be updated to reference oam-approval-protocol {$request.body#oam-approval-protocol}://...

File: ApplicationPattern.yaml [hunk 6]
```diff
@@ -2191,6 +2310,7 @@ paths:
                 topology-operation-fc-update: '/v1/update-fc'
                 topology-operation-fc-port-update: '/v1/update-fc-port'
                 topology-operation-fc-port-deletion: '/v1/delete-fc-port'
+                topology-application-protocol: 'HTTP'
```
Comment (IswaryaaS) [inline ApplicationPattern.yaml:2313]: uri definitions for callbacks of /v1/redirect-topology-change-information has protocol https which needs to be updated to reference topology-application-protocol {$request.body#topology-application-protocol}://...

**Review by IswaryaaS — CHANGES_REQUESTED** [review]

File: ApplicationPattern.yaml (hunk 2 above)
Comment (openBackhaul) [reply ApplicationPattern.yaml:639]: Corrected as proposed. Many thanks for indicating.

File: ApplicationPattern.yaml (hunk 4 above)
Comment (openBackhaul) [reply ApplicationPattern.yaml:1400]: Changed as proposed. Many thanks for the concrete proposal.

File: ApplicationPattern.yaml (hunk 5 above)
Comment (openBackhaul) [reply ApplicationPattern.yaml:1734]: Changed as proposed. Many thanks for the concrete proposal.

**Comment by openBackhaul** [comment]
After replacing _https_ by _[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol], the YAML didn't successfully validate in Postman API editor anymore.
Decided to substitute _https_ by 
````
url: #[/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-c-bm-ro-2-0-0-000/layer-protocol=0/tcp-client-interface-1-0:tcp-client-interface-pac/tcp-client-interface-configuration/remote-protocol]://[...]
````

File: ApplicationPattern+data.json [hunk 7]
```diff
@@ -23,7 +23,7 @@
                 },
                 "action-profile-configuration": {
                   "request": {
-                    "protocol-reference": "/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-s-bm-001/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/protocol",
+                    "protocol-reference": "/core-model-1-4:control-construct/logical-termination-point=*-1-0-0-tcp-s-bm-001/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol",
```
Comment (IswaryaaS) [inline ApplicationPattern+data.json:26]: we have uuid in profile section starting with prefix * and xx, which will be replaced by the actual application-name's notation in individual application. Here, uuid prefix is * in profile section whereas xx in logical-termination-point list. please let me know if we can maintain a common pattern for denoting the application-name holder (either * or xx) to maintain hormony.

Comment (openBackhaul) [reply ApplicationPattern+data.json:26]: Made a CTRL+h from *-1-0-0 to xx-1-0-0. Thanks for indicating.

**Review by IswaryaaS — APPROVED** [review]

**Pull request merged**